products/compute/api.yaml

```diff
@@ -8023,6 +8023,127 @@ objects:
                     - :NEVER
                     - :ON_PERMANENT_INSTANCE_DELETION
                   default_value: :NEVER
+  - !ruby/object:Api::Resource
+    name: 'RegionPerInstanceConfig'
+    base_url:  'projects/{{project}}/regions/{{region}}/instanceGroupManagers/{{instance_group_manager}}'
+    min_version: beta
+    description: |
+      A config defined for a single managed instance that belongs to an instance group manager. It preserves the instance name
+      across instance group manager operations and can define stateful disks or metadata that are unique to the instance.
+      This resource works with regional instance group managers.
+    create_verb: :POST
+    create_url: projects/{{project}}/regions/{{region}}/instanceGroupManagers/{{instance_group_manager}}/createInstances
+    update_verb: :POST
+    update_url: projects/{{project}}/regions/{{region}}/instanceGroupManagers/{{instance_group_manager}}/updatePerInstanceConfigs
+    delete_verb: :POST
+    delete_url: projects/{{project}}/regions/{{region}}/instanceGroupManagers/{{instance_group_manager}}/deletePerInstanceConfigs
+    read_verb: :POST
+    self_link: projects/{{project}}/regions/{{region}}/instanceGroupManagers/{{instance_group_manager}}/listPerInstanceConfigs
+    identity:
+      - name
+    nested_query: !ruby/object:Api::Resource::NestedQuery
+      keys:
+        - items
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation': 'https://cloud.google.com/compute/docs/instance-groups/stateful-migs#per-instance_configs'
+      api: 'https://cloud.google.com/compute/docs/reference/rest/beta/instanceGroupManagers'
+    async: !ruby/object:Api::OpAsync
+      operation: !ruby/object:Api::OpAsync::Operation
+        kind: 'compute#operation'
+        path: 'name'
+        base_url: 'projects/{{project}}/regions/{{region}}/operations/{{op_id}}'
+        wait_ms: 1000
+        timeouts: !ruby/object:Api::Timeouts
+          insert_minutes: 15
+          update_minutes: 6
+          delete_minutes: 15
+      result: !ruby/object:Api::OpAsync::Result
+        path: 'targetLink'
+      status: !ruby/object:Api::OpAsync::Status
+        path: 'status'
+        complete: 'DONE'
+        allowed:
+          - 'PENDING'
+          - 'RUNNING'
+          - 'DONE'
+      error: !ruby/object:Api::OpAsync::Error
+        path: 'error/errors'
+        message: 'message'
+    parameters:
+      - !ruby/object:Api::Type::ResourceRef
+        name: 'region'
+        resource: 'Region'
+        imports: 'name'
+        description: |
+          Region where the containing instance group manager is located
+        required: true
+        url_param_only: true
+        input: true
+      - !ruby/object:Api::Type::ResourceRef
+        name: 'instanceGroupManager'
```

Comment:
Should this be `regionInstanceGroupManager`?

Reply:
i like being explicit, lets say yes for name. I'm not sure it matters for the actual resource being ref'd since I think providing a partial/full URI to the RIGM will break it (URL params and mutex don't use expanders), but probably good for consistency.